agent/consul/catalog_endpoint.go

```diff
@@ -125,6 +125,13 @@ func (c *Catalog) Register(args *structs.RegisterRequest, reply *struct{}) error
 			check.Node = args.Node
 		}
 		checkPreApply(check)
+
+		// Populate check type for cases when this is a direct reg with /catalog/register
+		// Typically this is populated when a check is registered with an agent
```

Comment:
What I'm trying to communicate is that the user registered the check directly to the catalog. Anti-entropy is different because of the indirection. Users add checks to the agent then the agent triggers a sync. 

How about:

```go
// Populate check type for cases when a check is registered in the catalog directly and not via anti-entropy
```